### ISSUE TITLE
[Feature] Replace the `data-` attributes usage with a shorter syntax

### ISSUE THREAD

**Issue by titouanmathis** [studiometa/js-toolkit#492]
Repeating `data-component` and `data-option-...` for each component can be tedious and surcharge the HTML templates, adding complexity that might not be needed. 

This issue will try to explore a replacement API for the current one. 

- [ ] List the HTML attributes API
- [ ] Benchmark other tools: Vue, Alpine, HTMX, etc. 
- [ ] Define a new API and how it could be implemented

**Comment by titouanmathis** [comment]
A first proposal:

|   Current HTML API  | Suggested HTML API | Fallback if conflict with other attributes |
|---------------------|--------------------|--------------------------------------------|
| data-component      | is                 | :is                                        |
| data-option-name    | name               | :name                                      |
| data-option-state   | state              | :state                                     |
| data-option-is-open | is-open            | :is-open                                   |
| data-ref            | ref                | :ref                                       |

**Example**

<table><thead><tr><th>
Current
</th><th>
Suggested
</th></tr></thead><tbody><tr><td>

```html
<button
  type="button"
  data-component="Action"
  data-option-target="Panel Modal"
  data-option-selector=".can-be-closed"
  data-option-method="close">
  Click me
</button>
```

</td><td>

```html
<button
  type="button"
  is="Action"
  target="Panel Modal"
  selector=".can-be-closed"
  method="close">
  Click me
</button>
```


</td></tr></tbody></table>

**Pros**

- Shorter syntax, simpler API
- Option attributes can be merged with existing attribute, i.e. name for an input

**Cons**

- The `is` attribute is reserved ([doc](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/is)
- Attributes do not pass the W3C validator 
- Attributes might conflict with other ones

**Comment by titouanmathis** [comment]
This issue can be considered resolved with #495 which add support for configuration for the attributes used for each HTML API: component, options and refs.
